app/controllers/concerns/phone_confirmation_flow.rb

```diff
@@ -72,11 +72,10 @@ def send_confirmation_code
     # user's session. Re-sending the confirmation code doesn't generate a new one.
     self.confirmation_code = generate_confirmation_code unless confirmation_code
 
-    if current_otp_method == :voice
-      VoiceSenderOtpJob.perform_later(confirmation_code, unconfirmed_phone)
-    else
-      SmsSenderOtpJob.perform_later(confirmation_code, unconfirmed_phone)
-    end
+    job = "#{current_otp_method.to_s.capitalize}SenderOtpJob".constantize
+
+    job.perform_later(confirmation_code, unconfirmed_phone)
+
```

Comment:
Generating references at runtime using `.constantize` is also pretty horrible though :(   It makes it really hard to `grep` for references and understand there things are using the codebase.   I wish there was a nicer solution.